lib/src/components/NxVulnerabilityIndicator/NxVulnerabilityIndicator.scss

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+ @use '../../scss-shared/nx-dark-mode-helpers';
+ 
+// eyeballed, assumes $nx-font-size surrounding text
+$-vertical-alignment: -0.1em;
+
+.nx-vulnerability-indicator {
+  font-size: var(--nx-font-size-s);
```

Comment:
The visible shape of this fontawesome icon doesn't take up the full height of its `<svg>` box: the `<svg>` is 14px tall but the visible square is only 12.5px tall.  As it happens, this means that if you use the regular 16px font size, the visible square comes out to the desired 14px.  And then, since the "font size" of the icon is the same as that of the adjacent text, we don't have to use a custom `vertical-align`.

Reply:
> The visible shape of this fontawesome icon doesn't take up the full height of its `<svg>` box: the `<svg>` is 14px tall but the visible square is only 12.5px tall whereas it's supposed to be 14px. As it happens, this can be fixed by using the regular 16px font size, causing the visible square to come out to the desired 14px. And then, since the "font size" of the icon is the same as that of the adjacent text, we don't have to use a custom `vertical-align`.

Hmm currently discussing this with Kiran will get back to you,
but I think the default font awesome icon svg size determined by font-size should match the threat indicator 
so we can keep the desired visual weight balance benefit
(For example: if the circle without the container is exactly the same dimension as the square, it would be visually smaller. fa intentionally made the circle overshoot so it is visually balanced with the other shapes)

<img width="586" alt="Screenshot 2024-10-15 at 12 27 30 PM" src="https://github.com/user-attachments/assets/54e3c7bb-68e2-4da9-9131-414ad5a20677">

Reply:
Just had a discussion with Kiran, and yes this is what we want.
Kiran is updating the mockup to use fa-square and there's going to be a wider discussions on how fa are imported in Figma to keep the boundary so the intended relative sizing is consistent.

I would think the (eyeballing) vertical alignment will be similar as well since they both have now the same relative alignment but will additionally confirm that with design if they are satisfied.

Reply:
> Just had a discussion with Kiran, and yes this is what we want. Kiran is updating the mockup to use fa-square and there's going to be a wider discussions on how fa are imported in Figma to keep the boundary so the intended relative sizing is consistent.
> 
> I would think the (eyeballing) vertical alignment will be similar as well since they both have now the same relative alignment but will additionally confirm that with design if they are satisfied.

We added sidebearing to the icon so it aligns with NxThreatIndicator:
[70aa706](https://github.com/sonatype/sonatype-react-shared-components/pull/1199/commits/70aa706afe3e9c60f8784673aecdbd2b5422c094)

<img width="305" alt="Screenshot 2024-10-15 at 5 21 05 PM" src="https://github.com/user-attachments/assets/d8844efd-d258-475a-abb4-e0fd1f562dbd">

Reply:
> We added sidebearing to the icon so it aligns with NxThreatIndicator:

Is that a requirement? In what circumstance would a threat indicator and a vulnerability indicator be used together like this where their relative alignment is relevant?

> Kiran is updating the mockup to use fa-square and there's going to be a wider discussions on how fa are imported in Figma to keep the boundary so the intended relative sizing is consistent.

I've yet to find a way to get the icon name out of figma, but at any rate, the rounded box that I see there is still 14px tall.

Reply:
> Is that a requirement? In what circumstance would a threat indicator and a vulnerability indicator be used together like this where their relative alignment is relevant?

It would be unlikely for these two be used together,
but it it was done for consistency sake.

> I've yet to find a way to get the icon name out of figma, but at any rate, the rounded box that I see there is still 14px tall.

Yes, I believe this is because they have not been properly imported into Figma.
We're going to have a wider discussion on this with design
but some of the fa icons imported in Figma have their boundary box discarded (like the em square in type)
so its just the shape that's scaled to 14px.

So in this case the rounded box should be 12.25px within a boundary box that is 12.25px by 14px
but because the boundary box is discarded,
it's left with the 12.25px square scaled to 14px becoming 14x14px in Figma.

I've talked about this with Kiran and I think it's still being worked on the current mockup.
But the fix has been agreed upon.

Reply:
> > We added sidebearing to the icon so it aligns with NxThreatIndicator:
> 
> Is that a requirement? In what circumstance would a threat indicator and a vulnerability indicator be used together like this where their relative alignment is relevant?
> 
> > Kiran is updating the mockup to use fa-square and there's going to be a wider discussions on how fa are imported in Figma to keep the boundary so the intended relative sizing is consistent.
> 
> I've yet to find a way to get the icon name out of figma, but at any rate, the rounded box that I see there is still 14px tall.

Had a meeting with Kiran and we addressed the icon issue here for this component:
Kiran said that this is in the process of moving this to Kaleido: along with the rest:
https://www.figma.com/design/jG8RQhjPyIbisYAxDlOyt7/Vulnerability-Threat-Level-Count?node-id=265-213&node-type=frame&t=74prgOHcbyzxWgup-0